core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,142 @@
+        input=([(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands", "new_bands")]),
+        output_get_features=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "bands"), (FeatureType.MASK, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+    ),
+    ParsingTestClass(
+        input=(
+            {
+                FeatureType.DATA: [("bands", "new_bands"), ("CLP", "new_CLP")],
+                FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")],
+            }
+        ),
+        output_get_features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        output_get_renamed_features=[
+            (FeatureType.DATA, "bands", "new_bands"),
+            (FeatureType.DATA, "CLP", "new_CLP"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        output_get_feature_specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+
+TEST_CASES_ELLIPSIS = [
+    ParsingTestClass(
+        input=({FeatureType.DATA: ..., FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")]}),
+        output_get_features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        output_get_renamed_features=[
+            (FeatureType.DATA, "bands", "bands"),
+            (FeatureType.DATA, "CLP", "CLP"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        output_get_feature_specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_EMPTY + TEST_CASES)
+def test_FeatureParser(test_case: ParsingTestClass):
+    parser = FeatureParser(test_case.input)
+    assert parser.get_features() == test_case.output_get_features
+    assert parser.get_renamed_features() == test_case.output_get_renamed_features
+    assert parser.get_feature_specifications() == test_case.output_get_feature_specifications
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_ELLIPSIS)
+def test_FeatureParser_Error(test_case: ParsingTestClass):
+    parser = FeatureParser(test_case.input)
+    with pytest.raises(ValueError):
+        parser.get_features()
+    with pytest.raises(ValueError):
+        parser.get_renamed_features()
+
+
+@pytest.fixture(name="eopatch")
+def eopatch_fixture():
```

Comment:
the eopatch should have at least one of BBOX and TIMESTAMP, to make sure they are handled properly

Reply:
done

Reply:
it's good that some cases have duplicate names (so a FeatureType.DATA and FeatureType.MASK with the same name) as it is a weird but plausible case. But you have a bit too much duplication

Reply:
done